src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1719,6 +1591,56 @@ internal static Process Start(ProcessStartInfo startInfo, ProcessWindowStyle win
             return processToReturn;
         }
 
+        private void ShellExecuteFunction() {
+            if (!NativeMethods.ShellExecuteEx(_executeInfo))
+            {
+                _errorCode = Marshal.GetLastWin32Error();
+                if (_errorCode == 0)
+                {
+                    switch ((long)_executeInfo.hInstApp)
+                    {
+                        case NativeMethods.SE_ERR_FNF: _errorCode = NativeMethods.ERROR_FILE_NOT_FOUND; break;
+                        case NativeMethods.SE_ERR_PNF: _errorCode = NativeMethods.ERROR_PATH_NOT_FOUND; break;
+                        case NativeMethods.SE_ERR_ACCESSDENIED: _errorCode = NativeMethods.ERROR_ACCESS_DENIED; break;
+                        case NativeMethods.SE_ERR_OOM: _errorCode = NativeMethods.ERROR_NOT_ENOUGH_MEMORY; break;
+                        case NativeMethods.SE_ERR_DDEFAIL:
+                        case NativeMethods.SE_ERR_DDEBUSY:
+                        case NativeMethods.SE_ERR_DDETIMEOUT: _errorCode = NativeMethods.ERROR_DDE_FAIL; break;
+                        case NativeMethods.SE_ERR_SHARE: _errorCode = NativeMethods.ERROR_SHARING_VIOLATION; break;
+                        case NativeMethods.SE_ERR_NOASSOC: _errorCode = NativeMethods.ERROR_NO_ASSOCIATION; break;
+                        case NativeMethods.SE_ERR_DLLNOTFOUND: _errorCode = NativeMethods.ERROR_DLL_NOT_FOUND; break;
+                        default: _errorCode = (int)_executeInfo.hInstApp; break;
+                    }
+                }
+                _succeeded = false;
+            }
+        }
+
+        private bool ExecuteOnSTAThread()
+        {
+            if (Thread.CurrentThread.GetApartmentState() != System.Threading.ApartmentState.STA)
+            {
+                ThreadStart threadStart = new ThreadStart(this.ShellExecuteFunction);
+                Thread thread = new Thread(threadStart);
+                thread.SetApartmentState(System.Threading.ApartmentState.STA);
+                thread.Start();
+                thread.Join();
+            }
+            else
+            {
+                ShellExecuteFunction();
+            }
+            return _succeeded;
```

Comment:
`_succeeded` is never set to true. Maybe you can set its default value to true, or you can use similar way as in https://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/Process.cs,2850